src/main/service/listManipulation/magsStandardListService.ts

```diff
@@ -0,0 +1,92 @@
+import moment from 'moment-timezone';
+import {DataManipulationService} from '../dataManipulationService';
+import {DateTimeHelper} from '../../helpers/dateTimeHelper';
+
+const dataManipulationService = new DataManipulationService();
+const dateTimeHelper = new DateTimeHelper();
+
+export class MagsStandardListService {
+  public manipulatedMagsStandardListData(magsStandardListData: object, language: string, languageFile: string): object {
+    let hearingCount = 0;
+    magsStandardListData['courtLists'].forEach(courtList => {
+      courtList['courtHouse']['courtRoom'].forEach(courtRoom => {
+        courtRoom['session'].forEach(session => {
+          session['sittings'].forEach(sitting => {
+            const allHearings = [];
+            session['formattedJudiciaries'] = dataManipulationService.findAndManipulateJudiciary(sitting);
+            this.formatCaseTime(sitting, 'h:mma');
+            hearingCount = hearingCount + sitting['hearing'].length;
+            sitting['hearing'].forEach(hearing => {
+              sitting['formattedDuration'] = dateTimeHelper.formatDuration(sitting['durationAsDays'] as number,
+                sitting['durationAsHours'] as number, sitting['durationAsMinutes'] as number, language, languageFile);
+              hearing.party.forEach(party => {
+                const hearingString = JSON.stringify(hearing);
+                const hearingObject = JSON.parse(hearingString);
+                if (hearingObject?.party) {
+                  this.manipulatePartyInformation(hearingObject, party);
+                  hearingObject['case'].forEach(thisCase => {
+                    thisCase['formattedConvictionDate'] = dateTimeHelper.formatDate(thisCase['convictionDate'], 'DD/MM/YYYY');
+                    thisCase['formattedAdjournedDate'] = dateTimeHelper.formatDate(thisCase['adjournedDate'], 'DD/MM/YYYY');
+                  });
+                  allHearings.push(hearingObject);
+                }
+              });
+            });
+            sitting['hearing'] = allHearings;
+          });
+        });
+        courtRoom['totalHearing'] = hearingCount;
+        hearingCount = 0;
+      });
+    });
+
+    return magsStandardListData;
+  }
+
+  private formatCaseTime(sitting: object, format: string): void {
+    if (sitting['sittingStart'] !== '') {
+      sitting['time'] = moment.utc(sitting['sittingStart']).tz(dataManipulationService.timeZone).format(format);
+    }
+  }
+
+  private manipulatePartyInformation(hearing: any, party: any): void {
+    let defendant = '';
+    switch (DataManipulationService.convertPartyRole(party.partyRole)) {
```

Comment:
Swicth is normally used for 3 cases or more. Can we use if statement here instead?

Reply:
fixed.